src/utils.ts

```diff
@@ -266,6 +274,176 @@ export const getConfigFilesChangedInPR = async (pr: devOps.PullRequestProperties
+  return readmeUrl.substring(readmeUrl.indexOf("specification"), readmeUrl.indexOf("readme.md"));
+}
+
+export const isTagExisting = (config:string,tag:string):boolean => {
+    const content = fs.readFileSync(config, { encoding: "utf8" });
+    const readme = parse(content);
+   return getInputFilesForTag(readme.markDown,tag) != undefined
+} 
+
+/**
+ * @return {map} ,key is the readme and value is an array of the changed files belonging to the readme  
+ * [["1/readme.md",["file1.json","file2.json"]]] 
+ * @param filesChanged
+ */
+export const getChangeFilesReadmeMap = async (
+         filesChanged: string[]
+       ): Promise<Map<string, string[]>> => {
+         const configFiles = new Map<string, string[]>();
+         for (let fileChanged of filesChanged) {
+           const backup = fileChanged;
+           while (fileChanged.startsWith("specification")) {
+             if (fileChanged.toLowerCase().endsWith("readme.md") && fs.existsSync(fileChanged)) {
+               if (configFiles.has(fileChanged)) {
+               } 
+               else {
+                 configFiles.set(fileChanged, []);
+               }
+               if (!backup.endsWith("readme.md")) {
+                 let value = configFiles.get(fileChanged);
+                 if (value) {
+                   value.push(backup);
+                 }
+               }
+               break;
+             }
+             // select parent readme
+             const parts = fileChanged.split("/");
+             parts.pop();
+             parts.pop();
+             parts.push("readme.md");
+             fileChanged = parts.join("/");
+           }
+         }
+         return configFiles;
+       };
+
+/**
+ * return [["Sevice/readme.md",["tag1","tag2"]]...]
+ * the tags are sorted by refered changed file's count 
+ */
+export const getTagsFromChangedFile = async (
+         filesChanged: string[]
+       ): Promise<Map<string, string[]>> => {
+  try {
+    console.log(filesChanged);
+    // traverse up to readme.md files
+    const configFiles = await getChangeFilesReadmeMap(filesChanged);
+
+    const tagsAffectedMap = new Map<string, string[]>();
+
+    configFiles.forEach((changedFiles, key) => {
+      const content = fs.readFileSync(key, { encoding: "utf8" });
+      const readme = parse(content);
+      const relativePath = getReadMeRelativePathToRepoWithoutFileName(key);
+      /**
+      *  count the changed file count for each tags
+      */
+      const tagsCnt = new Map<string, number>();
+      changedFiles.map((changedFile) => {
+        const tags = getTagsForFilesChanged(readme, [
+          changedFile.substring(changedFiles.indexOf(relativePath)),
+        ]);
+        tags.forEach((element) => {
+          const oldCnt = tagsCnt.get(element);
+          tagsCnt.set(element, oldCnt ? oldCnt + 1 : 1);
+        });
+      });
+      
+      /**
+       * first sort by count, then by tag
+       */
+      const sortedTagsCnt = [...tagsCnt].sort((a, b) => {
```

Comment:
Does the index 0,1 means key and value of object?

Reply:
correct